### PR TITLE
Fix missing errors for indexes

### LINKED ISSUE #300: [PE-D][Tester E] Unhandled out-of-bounds exception for assign-task method
When entering the command "assign-task 1 0", the exception is not handled within the app but instead throws an exception in the console.

Step to recreate:
1. Enter the command "assign-task 1 0" assuming the person at index 1 is a vendor.

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: kohkakohla/ped#1

### LINKED ISSUE #302: [PE-D][Tester D] When deleting negating indices, no error message is shown
Input: ` delete-task -1`

### Issue
Error message not shown when deleting negative indices. Given someone did not read your UG well and entered these commads, they might feel a little lost when nothing happens when they enter a negative index.

### Proposed Solution
Output an error to the user stating that negative indices are not allow for `delete-task` command. Users will be more clear on the mistake they have made!

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: bigismols/ped#10

### LINKED ISSUE #308: [PE-D][Tester E] Missing error message for mark-task command
When I get an error message after inputting a floating point number (eg. mark-task 2.5) as an index, I get the above missing error message.

Steps to recreate:
1. Run the command "mark-task 2.5"

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: kohkakohla/ped#5

### PR DESCRIPTION
**Pull request by riccoljy**
Fixes #308 fixes #302 fixes #300 
Also simplified as per IDE suggestion to assertEquals and assertNotEquals

### CODE REVIEW

**Review by tingxuanp — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/task/AssignTaskCommand.java [hunk 1]
```diff
@@ -71,10 +71,9 @@ public CommandResult execute(Model model) throws CommandException {
         List<Task> lastShownTaskList = model.getFilteredTaskList();
 
 
-        if (personIndex.getZeroBased() >= lastShownPersonList.size()) {
-            throw new CommandException(String.format(
-                    Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX, 1 , lastShownPersonList.size()
-            ));
+        if (personIndex.getZeroBased() >= lastShownPersonList.size() || personIndex.getZeroBased() < 0) {
```
Comment (HanB1n) [inline src/main/java/seedu/address/logic/commands/task/AssignTaskCommand.java:74]: LGTM! But in the assign-wedding the same error exists.

**Review by HanB1n — CHANGES_REQUESTED** [review]
LGTM!

**Review by HanB1n — APPROVED** [review]
LGTM

**Review by HanB1n — APPROVED** [review]

**Pull request merged**